The-Puffin-Room/src/components/CommentCard.jsx

```diff
@@ -1,27 +1,42 @@
 import { LoginContext } from "../contexts/LoginContext";
-import { useContext } from "react";
+import { useContext, useState } from "react";
+
+import { deleteComment } from "../api";
 
 import dateFormat from "dateformat";
-import moment from 'moment';
+import moment from "moment";
 moment().format();
 
-function CommentCard({ comment }) {
-const { loggedUser } = useContext(LoginContext);
+function CommentCard({ comment , onDelete}) {
+  const date = dateFormat(comment.created_at, "isoDate");
+  const howLongAgo = moment(date, "YYYYMMDD").fromNow();
 
-const date = dateFormat(comment.created_at, "isoDate");
-const howLongAgo= moment(date, "YYYYMMDD").fromNow();
+  const { loggedUser } = useContext(LoginContext);
 
-if(!comment.author){
-  comment.author = loggedUser.username
-}
+  if (!comment.author) {
+    comment.author = loggedUser.username;
+  }
+
+  const handleClick = () => {
+    console.log(comment.comment_id)
+    console.log("clicked")
+    onDelete(comment.comment_id)
+    deleteComment(comment.comment_id)
```

Comment:
we should also undo the optimistic render if the comment fails to be removed from the db